test/rubygems/test_gem_indexer.rb

```diff
@@ -179,6 +190,8 @@ def test_generate_index_modern_back_to_back
 
     assert_indexed @indexerdir, "latest_specs.#{@marshal_version}"
     assert_indexed @indexerdir, "latest_specs.#{@marshal_version}.gz"
+  ensure
+    FileUtils.rm_rf(indexer.directory)
```

Comment:
If something failed before the `indexer` variable is defined, this would result in a runtime error in `ensure`, right? Really loving to be able to use `ensure` without explicit begin, but in this case the correct thing would be to only wrap the code after `indexer` is defined?